deepspeed/runtime/zero/constants.py

```diff
@@ -116,7 +116,8 @@
 
 # gathers params for saving a model - inefficient but is required in certain situations
```

Comment:
Could someone kindly help explain which situations were required by this 16bit parameters gathering (infeeicient) feature, given that there is zero_to_fp32.py script which can help save the parameters?  Thanks.

Reply:
This is for those who can't be bothered with running `zero_to_fp32.py` and want the 16-bit model extracted on the fly - which is fine for tiny to small models but very slow for large models.

It's also the default in the HF Trainer integration of Deepspeed to make it easy for users to start and have things work transparently. But the documentation explains how to improve upon this default.
https://huggingface.co/docs/transformers/main/main_classes/deepspeed#getting-the-model-weights-out